### PR TITLE
tests: Add an nmstatectl integration test for 'set'

### PR DESCRIPTION
**Pull request by EdDev**
Configuring a static IPv4 address on an interface.

### CODE REVIEW

File: tests/integration/testlib/cmd.py [hunk 1]
```diff
@@ -20,7 +20,7 @@
 import subprocess
 
 
-def exec_cmd(cmd, env=None):
+def exec_cmd(cmd, env=None, stdin=None):
```
Comment (tyll) [inline tests/integration/testlib/cmd.py]: How about moving this into a separate commit? It is a prerequisite for the other change but not really related.

File: tests/integration/nmstatectl_test.py [hunk 2]
```diff
@@ -73,5 +88,12 @@ def test_show_command_with_yaml_format():
...
 
 
+def test_set_command_with_yaml_format():
+    ret = libcmd.exec_cmd(SET_CMD, stdin=ETH1_YAML_CONFIG)
+    rc, out, err = ret
+
+    assert_rc(rc, RC_SUCCESS, ret)
+
+
 def assert_rc(actual, expected, return_tuple):
```
Comment (tyll) [inline tests/integration/nmstatectl_test.py]: IMHO it is strange that the same value is passed twice to the method, once as actual and once in return_tuple.

**Comment by tyll** [comment]
The new test fails in the standard-ci.

File: tests/integration/nmstatectl_test.py (hunk 2 above)
Comment (EdDev) [reply tests/integration/nmstatectl_test.py]: This function does not know that the `expected` is in the `return_tuple`. I do not see anything nicer here, expect perhaps to use a separate function that always creates the string on input at the caller.

Anyway, it is not part of this PR.

File: tests/integration/testlib/cmd.py (hunk 1 above)
Comment (EdDev) [reply tests/integration/testlib/cmd.py]: We introduce it with the usage and it is not that complex. I find it simpler to review this way.
But if you insist, I will comply.

File: tests/integration/nmstatectl_test.py (hunk 2 above)
Comment (tyll) [reply tests/integration/nmstatectl_test.py]: You are right, it is not part of this PR. However, I do not follow what you are writing. The method is called `assert_rc` and it assumes that the return code is the first value of `return_tuple`. At the same time `actual` implies that it is the return code from `return_tuple`. I guess the method needs a doc string to explain when `actual =! return_tuple[0]` because it does not make sense to me to call it this way.

File: tests/integration/nmstatectl_test.py [hunk 3]
```diff
@@ -49,6 +50,20 @@
...
   mtu: 65536"""
 
+ETH1_YAML_CONFIG = b"""interfaces:
+- name: eth1
+  state: up
+  type: ethernet
+  mtu: 1500
+  ipv4:
+    address:
+    - ip: 192.168.122.250
```
[the remark this answers is not in the document]
Comment (EdDev) [reply tests/integration/nmstatectl_test.py]: Done

File: tests/integration/nmstatectl_test.py (hunk 2 above)
Comment (EdDev) [reply tests/integration/nmstatectl_test.py]: I will try to name and structure it better in a following patch.

File: tests/integration/testlib/cmd.py (hunk 1 above)
Comment (EdDev) [reply tests/integration/testlib/cmd.py]: In production code, we can "show" how a new unit is used using a test. In tests, when introducing a new "service" option, it is convenient to "show" how to use it or assure it really works by using it (in the same patch).

There are however cases where the  size of the new "service" or test that uses it is too big to make sense, there separation is preferred. Also, when introducing the new "service", only one needs to use it in the same commit, the other (if there are any), better be changed in a separate commit/s.

Again, because it is a grey area, I am fine with either ways, as long as they are merged together (with gerrit, it is harder to assure they get merged together).

**Comment by tyll** [comment]
required for #227

File: tests/integration/nmstatectl_test.py (hunk 2 above)
Comment (tyll) [reply tests/integration/nmstatectl_test.py]: Thank you!

File: tests/integration/nmstatectl_test.py [hunk 4]
```diff
@@ -99,5 +114,12 @@ def test_show_command_only_non_existing():
     assert len(state[Constants.INTERFACES]) == 0
 
 
+def test_set_command_with_yaml_format():
+    ret = libcmd.exec_cmd(SET_CMD, stdin=ETH1_YAML_CONFIG)
+    rc, out, err = ret
```
Comment (tyll) [inline tests/integration/nmstatectl_test.py]: just a minor nitpick, IMHO it is not obvious what `err` means. It seems to be `stderr` but I would not assume this just based on the variable name. Using `stdout` and `stderr` here would be more obvious since the input to `exec_cmd` is also `stdin`.

Comment (EdDev) [reply tests/integration/nmstatectl_test.py]: IMO `stderr` and `stdout` may be used as either reserved keywords or just as package names, it is "too" common to use it like this.
On the other hand, it is used as an argument name by other existing function (including `exec_cmd`).

I don't really know, it feel uncomfortable for me to use std* names, perhaps due to my c/c++ history. We will then argue if this is a stream or a string and never end this discussion.
Leaving the coder to choose a reasonable name is a good middle ground IMO.
